src/gentropy/method/l2g/feature_factory.py

```diff
@@ -26,162 +26,121 @@ class ColocalisationFactory:
 
     @staticmethod
     def _get_max_coloc_per_credible_set(
+        colocalisation: Colocalisation,
         credible_set: StudyLocus,
         studies: StudyIndex,
-        colocalisation: Colocalisation,
-        colocalisation_method: str,
     ) -> L2GFeature:
         """Get the maximum colocalisation posterior probability for each pair of overlapping study-locus per type of colocalisation method and QTL type.
 
         Args:
+            colocalisation (Colocalisation): Colocalisation dataset
             credible_set (StudyLocus): Study locus dataset
             studies (StudyIndex): Study index dataset
-            colocalisation (Colocalisation): Colocalisation dataset
-            colocalisation_method (str): Colocalisation method to extract the max from
 
         Returns:
             L2GFeature: Stores the features with the max coloc probabilities for each pair of study-locus
-
-        Raises:
-            ValueError: If the colocalisation method is not supported
         """
-        if colocalisation_method not in ["COLOC", "eCAVIAR"]:
-            raise ValueError(
-                f"Colocalisation method {colocalisation_method} not supported"
-            )
-        if colocalisation_method == "COLOC":
-            coloc_score_col_name = "log2h4h3"
-            coloc_feature_col_template = "ColocLlrMaximum"
-
-        elif colocalisation_method == "eCAVIAR":
-            coloc_score_col_name = "clpp"
-            coloc_feature_col_template = "ColocClppMaximum"
+        colocalisation_df = colocalisation.df.select(
+            f.col("leftStudyLocusId").alias("studyLocusId"),
+            "rightStudyLocusId",
+            f.coalesce("log2h4h3", "clpp").alias("score"),
+            f.when(f.col("colocalisationMethod") == "COLOC", f.lit("Llr"))
```

Comment:
to avoid the hardcoding you might want to do something like:

```suggestion
            f.when(f.col("colocalisationMethod") == Coloc.METHOD_NAME, Coloc.STATISTIC_COLUMN)
```
These 2 attributes would belong to the `Coloc` method class. You would also use `Coloc.METHOD_NAME` to populate the original column `colocalisationMethod`. This would also reduce the risk of introducing errors as the logic will no longer depend on the semantics of the column (e.g. "COLOC").
 
Because these are properties of the method it would make sense. An abstract class for colocalisation methods might be overkill for only 2 methods and no prospect of expanding them, but I would be also ok with that. You could also build logic that iterates through all classes and build this when statements. Again, it could be an overkill in this context but it might be something you want to think about in other contexts (e.g. features, etc.).

Reply:
This is good to go.

Thinking about further improvements. There are probably 2 things that come to mind:
- Ensuring that all methods have METHOD_NAME, etc. using inheritance. Not urgent as we are not adding more methods short term.
- One class/function responsible for telling you the list of available colocalisation classes/methods. Inheritance of an abstract colocalisationMethod might help here as well. [Check this](https://stackoverflow.com/questions/3862310/how-to-find-all-the-subclasses-of-a-class-given-its-name). Then you could build your dictionary dynamically. On a separate note, I think we will need to add a similar strategy if we want to play with the features to include in L2G upon configuration.

Reply:
Thank you for the comments! I did a very similar [here](https://github.com/opentargets/gentropy/commit/dfaac7994760f0b76bb0a4fcfd494acd415e72f4) so I agree that we should implement this properly